core/server/proxy/src/main/java/alluxio/proxy/s3/signature/SignatureProcessor.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.proxy.s3.signature;
+
+import alluxio.proxy.s3.S3Exception;
+import alluxio.proxy.s3.auth.AWSAuthInfo;
+
+/**
+ * Parser to request auth parser for http request.
+ */
+public interface SignatureProcessor {
```

Comment:
I don't think you need an interface here. Are you expecting to have different `SignatureProcessor` for S3?